subsys/net/ip/tcp.h

```diff
@@ -79,7 +79,12 @@ enum net_tcp_state {
 /* RFC 1122 4.2.2.6 "If an MSS option is not received at connection
  * setup, TCP MUST assume a default send MSS of 536"
  */
-#define NET_TCP_DEFAULT_MSS   536
+/*#define NET_TCP_DEFAULT_MSS   536*/
+/* TODO: We don't yet implement full MSS negotiation in Zephyr,
+ * but have tests which rely on being able to send "full" IPv6
+ * (MTU 1280) packets, so for now set default MSS to higher value
+ */
+#define NET_TCP_DEFAULT_MSS   1280
```

Comment:
@pfl thanks, but:

1. What you say is pretty obvious and known to the author of the patch.
2. Zephyr IP stack has multiple violations of IP standards.
3. There's incremental work to make it compliant, and fixing one issue often leads to exposing issues in other parts of code.
4. This patch is part of the work to get more overall p.1 fixes into the 1.10 release, rather than less, which requires quite extensive negotiations and compromises among all the parties involves, see #119, #4821, #4934, #5015, etc.

Reply:
I can agree with points 1.-4. above. But in order to fix things, let's not break the ones that actually implement the standards properly.

Reply:
I am the author of MSS handling/negotiation support in Zephyr (there was none before I started to implement, I haven't finished yet). And for the sake of impeding 1.10 release, I find it acceptable to effectively disable the MSS handling functionality I added, because there're a lost of code in Zephyr which isn't ready for proper MSS handling (actually, only 2 pieces are guaranteedly ready: BSD Sockets, authors by me, and new HTTP API, for which @jukkar kindly accepted my suggestions).